data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java

```diff
@@ -70,18 +109,32 @@ public PostAggregateAndProofs(
         @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
       })
   @Override
-  public void handle(Context ctx) throws Exception {
-    try {
-      final SignedAggregateAndProof[] signedAggregateAndProofs =
-          parseRequestBody(ctx.body(), SignedAggregateAndProof[].class);
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
 
-      final SafeFuture<Optional<PostDataFailureResponse>> future =
-          provider.sendAggregateAndProofs(asList(signedAggregateAndProofs));
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final List<SignedAggregateAndProof> signedAggregateAndProofs = request.getRequestBody();
+    final SafeFuture<List<SubmitDataError>> future =
+        provider.sendAggregateAndProofs(signedAggregateAndProofs);
+
+    request.respondAsync(
+        future.thenApply(
+            errors -> {
+              if (errors.isEmpty()) {
+                return AsyncApiResponse.respondWithCode(SC_OK);
+              }
+
+              final ErrorListBadRequest data =
+                  ErrorListBadRequest.convert(PARTIAL_PUBLISH_FAILURE_MESSAGE, errors);
+              return AsyncApiResponse.respondWithObject(SC_BAD_REQUEST, data);
+            }));
+  }
 
-      handlePostDataResult(ctx, future);
-    } catch (IllegalArgumentException e) {
-      ctx.json(BadRequest.badRequest(jsonProvider, e.getMessage()));
-      ctx.status(SC_BAD_REQUEST);
-    }
+  private static DeserializableTypeDefinition<SignedAggregateAndProof> getRequestBodyType(
+      final SpecConfig specConfig) {
+    return new SignedAggregateAndProof.SignedAggregateAndProofSchema(specConfig)
+        .getJsonTypeDefinition();
```

Comment:
should be using `schemaDefinitions.getSignedAggregateAndProofSchema()` basically from schema definition cache i think?

Reply:
Updated now. Please check what I have done is what you meant